patch/gfx9/chip/llpcGfx9ConfigBuilder.cpp

```diff
@@ -2308,31 +2311,56 @@ Result ConfigBuilder::BuildPrimShaderRegConfig(
     }
     else
     {
-        // Without tessellation
-        const auto pPipelineInfo = static_cast<const GraphicsPipelineBuildInfo*>(pContext->GetPipelineBuildInfo());
-        const auto topology = pPipelineInfo->iaState.topology;
-        if (topology == VK_PRIMITIVE_TOPOLOGY_POINT_LIST)
-        {
-            gsOutputPrimitiveType = POINTLIST;
-        }
-        else if ((topology == VK_PRIMITIVE_TOPOLOGY_LINE_LIST) ||
-                 (topology == VK_PRIMITIVE_TOPOLOGY_LINE_STRIP) ||
-                 (topology == VK_PRIMITIVE_TOPOLOGY_LINE_LIST_WITH_ADJACENCY) ||
-                 (topology == VK_PRIMITIVE_TOPOLOGY_LINE_STRIP_WITH_ADJACENCY))
-        {
-            gsOutputPrimitiveType = LINESTRIP;
-        }
-        else if ((topology == VK_PRIMITIVE_TOPOLOGY_TRIANGLE_LIST) ||
-                 (topology == VK_PRIMITIVE_TOPOLOGY_TRIANGLE_STRIP) ||
-                 (topology == VK_PRIMITIVE_TOPOLOGY_TRIANGLE_FAN) ||
-                 (topology == VK_PRIMITIVE_TOPOLOGY_TRIANGLE_LIST_WITH_ADJACENCY) ||
-                 (topology == VK_PRIMITIVE_TOPOLOGY_TRIANGLE_STRIP_WITH_ADJACENCY))
+        // GS not present
+        if (hasTs)
```

Comment:
Indentation could be reduced here by one level by making it a chain: `if (hasGs) ... else if (hasTs) ... else ...`.

Reply:
Thank you. Will correct this. For subgroupInclusiveAdd(), I also find this problem months ago. It is regretable we cannot use methods in LLPC builder in patch passes due to current design.